src/test/java/org/sqlite/DBMetaDataTest.java

```diff
@@ -1669,4 +1667,593 @@ public void getPrimaryKeys(String table) throws SQLException {
+        assertThat(rs.next()).isFalse();
+        rsmeta = rs.getMetaData();
+        assertThat(rsmeta.getColumnName(1)).isEqualTo("PKTABLE_CAT");
+        assertThat(rsmeta.getColumnName(2)).isEqualTo("PKTABLE_SCHEM");
+        assertThat(rsmeta.getColumnName(3)).isEqualTo("PKTABLE_NAME");
+        assertThat(rsmeta.getColumnName(4)).isEqualTo("PKCOLUMN_NAME");
+        assertThat(rsmeta.getColumnName(5)).isEqualTo("FKTABLE_CAT");
+        assertThat(rsmeta.getColumnName(6)).isEqualTo("FKTABLE_SCHEM");
+        assertThat(rsmeta.getColumnName(7)).isEqualTo("FKTABLE_NAME");
+        assertThat(rsmeta.getColumnName(8)).isEqualTo("FKCOLUMN_NAME");
+        assertThat(rsmeta.getColumnName(9)).isEqualTo("KEY_SEQ");
+        assertThat(rsmeta.getColumnName(10)).isEqualTo("UPDATE_RULE");
+        assertThat(rsmeta.getColumnName(11)).isEqualTo("DELETE_RULE");
+        assertThat(rsmeta.getColumnName(12)).isEqualTo("FK_NAME");
+        assertThat(rsmeta.getColumnName(13)).isEqualTo("PK_NAME");
+        assertThat(rsmeta.getColumnName(14)).isEqualTo("DEFERRABILITY");
+
+        rs = meta.getPrimaryKeys(null, "nonexistent", null);
+        assertThat(rs).isNotNull();
+        assertThat(rs.next()).isFalse();
+        rsmeta = rs.getMetaData();
+        assertThat(rsmeta.getColumnName(1)).isEqualTo("TABLE_CAT");
+        assertThat(rsmeta.getColumnName(2)).isEqualTo("TABLE_SCHEM");
+        assertThat(rsmeta.getColumnName(3)).isEqualTo("TABLE_NAME");
+        assertThat(rsmeta.getColumnName(4)).isEqualTo("COLUMN_NAME");
+        assertThat(rsmeta.getColumnName(5)).isEqualTo("KEY_SEQ");
+        assertThat(rsmeta.getColumnName(6)).isEqualTo("PK_NAME");
+    }
+
+    @Nested
+    class DBMetadataTestWithAttachedDatabases {
+        File testDB;
+
+        @BeforeEach
+        public void init() throws IOException, SQLException {
+            testDB = Files.createTempFile("temp", ".sqlite").toFile();
+            stat.executeUpdate("attach database \"" + testDB.toURI().toURL() + "\" as db2;");
+            stat.executeUpdate(
+                    "create table db2.test2 (id integer primary key, fn float default 0.0, sn not null, intvalue integer"
+                            + "(5), realvalue real(8,3));");
+            stat.executeUpdate("create view db2.testView2 as select * from db2.test2;");
+        }
+
+        @Test
+        public void readSchemaWithAttachedDatabases() throws SQLException {
+            ResultSet rs = meta.getSchemas();
+            assertThat(rs.next()).isTrue();
+            ResultSetMetaData rsMeta = rs.getMetaData();
+            assertThat(rsMeta.getColumnCount()).isEqualTo(2);
+            assertThat(rsMeta.getColumnName(1)).isEqualTo("TABLE_SCHEM");
+            assertThat(rsMeta.getColumnName(2)).isEqualTo("TABLE_CATALOG");
+            assertThat(rs.getString("TABLE_SCHEM")).isEqualTo("db2");
+            assertThat(rs.next()).isTrue();
+            assertThat(rs.getString("TABLE_SCHEM")).isEqualTo("main");
```

Comment:
is there some ordering of the schemas mandated by JDBC ? or we do always return them in alphabetical order?

Reply:
```
"select name as TABLE_SCHEM, null as TABLE_CATALOG from pragma_database_list order by "
                                    + "TABLE_SCHEM;");
```
Ordering in SQL request

Reply:
Just noticed that:

> The results are ordered by TABLE_CATALOG and TABLE_SCHEM

In that case the tests should test for the ordering.